src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -4749,10 +4788,40 @@ else if (DataTypes.UNKNOWN_STREAM_LENGTH == dataLength)
                     }
                     currentColumn++;
                 }
+
+                // send this row, read its response (throw exception in case of errors) and reset command status
+                if (tdsWritterCached) {
+                    // TVP_END_TOKEN
+                    writeByte((byte) 0x00);
+
+                    writePacket(TDS.STATUS_BIT_EOM);
+
+                    TDSReader tdsReader = tdsChannel.getReader(command);
+                    int tokenType = tdsReader.peekTokenType();
+
+                    if (TDS.TDS_ERR == tokenType) {
+                        StreamError databaseError = new StreamError();
+                        databaseError.setFromTDS(tdsReader);
+
+                        SQLServerException.makeFromDatabaseError(con, null, databaseError.getMessage(), databaseError, false);
+                    }
+
+                    command.setInterruptsEnabled(true);
+                    command.setRequestComplete(false);
+                }
             }
         }
-        // TVP_END_TOKEN
-        writeByte((byte) 0x00);
+
+        // reset command status which have been overwritten
+        if (tdsWritterCached) {
+            command.setRequestComplete(false);
```

Comment:
Instead of resetting them to default, it will be safe to catch original value of these 3 variables and reset them. Also update these values in synchronized way just as in the existing code.